packages/flutter_tools/lib/src/android/gradle.dart

```diff
@@ -322,6 +323,9 @@ Future<void> buildGradleApp({
   if (androidBuildInfo.shrink) {
     command.add('-Pshrink=true');
   }
+  if (androidBuildInfo.buildInfo.dartDefines?.isNotEmpty ?? false) {
```

Comment:
Wondering if checking for null could mask a bug. I tried to make it so that `dartDefines` is not null for commands that accept it. There's even a check like this: https://github.com/flutter/flutter/blob/1ba4f1f509202f3d2ec16f053edcbe2df6b7107e/packages/flutter_tools/lib/src/commands/daemon.dart#L93

Reply:
Ah, nvm, I see now that you are changing this below.